### PR TITLE
fix(IDPay): [IODPAY-147,IODPAY-135,IODPAY-128] Refactor list item design

### PR DESCRIPTION
**Pull request by forrest57**
## Short description
refactor of `TimelineListItem.tsx` to follow its new design

## List of changes proposed in this pull request

- addition of new icon, as per design 
- addition of possible `greenLight` color to the `H4` component
- rework of the timelineListItem filtering system to allow for more variety
- the component now more strictly adheres to the new design
- update of Wallet and Timeline definitions
   - update of required tests
   - new map to generate the correct SVG icons
   - required update of timeline saga and `isLastPage` selector

DEPENDS ON https://github.com/pagopa/io-app/pull/4362

## How to test
Having one fully configured initiative with various timeline events, navigate to the initiative details' page, possibly also to the "show more" section, and verify that all the listItems render correctly.

### CODE REVIEW

File: ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx [hunk 1]
```diff
@@ -35,27 +39,54 @@ type TimelineOperationListItemProps = {
...
+    }
+    switch (operation.operationType) {
+      case TransactionOperationTypeEnum.TRANSACTION:
+        return <H4> {`–${formatNumberAmount(operation.amount, false)} €`}</H4>;
+      case TransactionOperationTypeEnum.REVERSAL:
+        return <H4> {`+${formatNumberAmount(operation.amount, false)} €`}</H4>;
+      case RefundOperationTypeEnum.PAID_REFUND:
+        return (
+          <H4 color="greenLight">
+            {" "}
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx:86]: What's the purpose of this whitespace?

Comment (forrest57) [reply ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx:86]: most likely it was generated by the formatter and went unnoticed along the way; pushing fix

**Review by dmnplb — APPROVED** [review]

File: ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { format } from "date-fns";
 import { ListItem } from "native-base";
 import React from "react";
-import { View, Image, StyleSheet } from "react-native";
+import { Image, StyleSheet, View } from "react-native";
 import { OperationTypeEnum as IbanOperationTypeEnum } from "../../../../../../definitions/idpay/timeline/IbanOperationDTO";
 import { OperationTypeEnum as OnboardingOperationTypeEnum } from "../../../../../../definitions/idpay/timeline/OnboardingOperationDTO";
 import { OperationListDTO } from "../../../../../../definitions/idpay/timeline/OperationListDTO";
 import { OperationTypeEnum as RefundOperationTypeEnum } from "../../../../../../definitions/idpay/timeline/RefundOperationDTO";
+import { OperationTypeEnum } from "../../../../../../definitions/idpay/timeline/RejectedInstrumentOperationDTO";
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx:9]:
```suggestion
import { OperationTypeEnum as  RejectedInstrumentOperationTypeEnum } from "../../../../../../definitions/idpay/timeline/RejectedInstrumentOperationDTO";
```

Maybe a bit too long but helps keeping code clear imho.

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/components/core/typography/H4.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { useTypographyFactory } from "./Factory";
 // when the weight is bold, only these color are allowed
 type AllowedBoldColors = Extract<
   IOColorType,
-  "bluegreyDark" | "blue" | "white"
+  "bluegreyDark" | "blue" | "white" | "greenLight"
```
Comment (francescopersico) [inline ts/components/core/typography/H4.tsx:10]: @dmnplb Is this safe?

**Review by francescopersico — CHANGES_REQUESTED** [review]

File: ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx [hunk 4]
```diff
@@ -37,27 +39,53 @@ type TimelineOperationListItemProps = {
...
-        return <Icon name={"error"} color="bluegreyLight" />;
+        return <Icon name={"warning"} color="red" />;
+
+      default:
+        if ("brandLogo" in operation) {
+          return (
+            <Image
+              style={styles.imageSize}
+              source={{ uri: operation.brandLogo }}
+            />
```
Comment (mastro993) [inline ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx:67]: We should start to use the new `IOLogoPaymentType` for payment methods icons. Look at the DS to know how to use them.

**Review by mastro993 — CHANGES_REQUESTED** [review]

File: ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx (hunk 2 above)
Comment (francescopersico) [reply ts/features/idpay/initiative/details/components/TimelineOperationListItem.tsx:9]: @forrest57 Can you fix this?

**Review by francescopersico — CHANGES_REQUESTED** [review]

**Review by francescopersico — APPROVED** [review]

**Pull request merged**